### PR TITLE
report: Skip data item if version doesn't exist

### PR DESCRIPTION
**Pull request by Raynos**
It's possible for some data from ncm-api to come back
as malformed. We do not want to include it in the report.

This allows `ncm-cli report` to return data from almost all
modules, instead of a single bad module in ncm-api causing the
entire report to fail.

If the version field is falsey then skip this item and drop it
from the report.

### CODE REVIEW

**Comment by Raynos** [comment]
Fixes https://github.com/nodesource/ncm-cli/issues/149

**Comment by brycebaril** [comment]
Spent 5 minutes looking into adding a test to this PR but it appears that the mock server uses version gating as part of its logic, so I aborted for now.

**Comment by Fishrock123** [comment]
This is wrong. Ignore it in the sort if you must.

**Comment by Fishrock123** [comment]
At least log **something**.  _Anything_.

**Comment by Raynos** [comment]
I've improved handling of bad data from ncm-api.

Instead of skipping the module we set the version to unknown.

The output of report looks like

```
raynos at raynos-Precision-5530  
~/nodesource/ncm-cli on handle-bad-data*
$ node bin/ncm-cli.js report --long --dir ./test/fixtures/poisoned-project/

╔═════════════════════════╗
║ poisoned-project Report ║
╚═════════════════════════╝

4 packages checked

  ! 0 critical risk
    1 high risk
    0 medium risk
    0 low risk

  ✓ No security vulnerabilities found

  ! 1 noncompliant modules found
    |➔ Run `ncm report --filter=compliance` for a list

─────────────────────────────────────────────────────────────────────────────────────────────────
  Modules
-------------------------------------------------------------------------------------------------
  Module Name                               Risk         License                 Security
┌──────────────────────────────────────────┬────────────┬───────────────────────┬───────────────┐
│ left-pad @ 1.3.0                         │ |||| High  │ X WTFPL               │ ✓ 0           │
│ is-path-in-cwd… @ 0.0.0-UNKNOWN-VERSION  │            │                       │               │
│ is-path-inside @ 2.1.0                   │ |||| None  │ ✓ MIT                 │ ✓ 0           │
│ path-is-inside @ 1.0.2                   │ |||| None  │ ✓ (WTFPL OR MIT)      │ ✓ 0           │
└──────────────────────────────────────────┴────────────┴───────────────────────┴───────────────┘
```

This reflects that we did find the module and we attempted to compute its score but we print empty because we have no data for that module.

File: test/lib/mock-packages.js [hunk 1]
```diff
@@ -5715,4 +5715,295 @@ module.exports =
       'data': null
     }
   ]
+},
+{
+  "name": "is-path-in-cwd",
```
Comment (Fishrock123) [inline test/lib/mock-packages.js]: This stuff all explodes `standard`. :/

File: test/report.js [hunk 2]
```diff
@@ -303,3 +304,15 @@ TestRunner.test('report output matches snapshot', (runner, t) =>
...
 )
+
+TestRunner.test('report with poisoned project', (runner, t) => {
+  let cmd = `report --long ${POISONED_PROJECT}`
+  runner.exec(cmd, (err, stdout, stderr) => {
+    t.equal(err.code, 1)
+    t.equal(stderr, '')
+    t.matchSnapshot(stdout, 'long-report-poisoned-output')
+    t.match(stdout, /poisoned-project Report/)
+    t.match(stdout, /left-pad @ 1.3.0/)
```
Comment (Fishrock123) [inline test/report.js]: This should probably also check for `is-path-in-cwd @ 0.0.0-UNKNOWN-VERSION`?

Comment (Fishrock123) [reply test/report.js]: Yep, add `t.match(stdout, /is-path-in-cwd… @ 0.0.0-UNKNOWN-VERSION/)`.

**Comment by kevinswiber** [comment]
@Fishrock123 I believe I resolved all the open issues while @Raynos is out.  PTAL.

**Comment by Fishrock123** [comment]
Looks good. I will land in the morning if it isn't already by then.

**Comment by Fishrock123** [comment]
Landed as https://github.com/nodesource/ncm-cli/commit/ea065b3cba869c9fed04d82271c6828c8f74a795

**Pull request closed without merging**